src/core/datatypes/phylogenetics/geography/RegionalFeatures.cpp

```diff
@@ -0,0 +1,280 @@
+        size_t time_index = it->first;
+        // loop over time-feature maps
+        for (auto jt = withinQuantitative[time_index].begin(); jt != withinQuantitative[time_index].end(); jt++) {
+            size_t feature_index = jt->first;
+            RegionalFeatureLayer tmp( it->first, jt->first, "within", "quantitative" );
+            std::vector<std::vector<double> > tmp_val;
+            tmp_val.push_back( withinQuantitative[time_index][feature_index] );
+            tmp.setFeatures(tmp_val);
+            feature_layers["within"]["quantitative"][time_index-1].push_back(tmp);
+        }
+    }
+    
+    for (auto it = betweenCategorical.begin(); it != betweenCategorical.end(); it++) {
+        feature_layers["between"]["categorical"].push_back( std::vector<RegionalFeatureLayer>() );
+        size_t time_index = it->first;
+        // loop over time-feature maps
+        for (auto jt = betweenCategorical[time_index].begin(); jt != betweenCategorical[time_index].end(); jt++) {
+            size_t feature_index = jt->first;
+            RegionalFeatureLayer tmp(time_index, feature_index, "between", "categorical" );
+            std::vector<std::vector<double> > tmp_val;
+            for (size_t i = 0; i < betweenCategorical[time_index][feature_index].size(); i++) {
+                tmp_val.push_back(std::vector<double>());
+                for (size_t j = 0; j < betweenCategorical[time_index][feature_index][i].size(); j++) {
+                    tmp_val[i].push_back((double)betweenCategorical[time_index][feature_index][i][j]);
+                }
+            }
+            tmp.setFeatures(tmp_val);
+            feature_layers["between"]["categorical"][time_index-1].push_back(tmp);
+        }
+    }
+    
+    for (auto it = betweenQuantitative.begin(); it != betweenQuantitative.end(); it++) {
+        feature_layers["between"]["quantitative"].push_back( std::vector<RegionalFeatureLayer>() );
+        size_t time_index = it->first;
+        // loop over time-feature maps
+        for (auto jt = betweenQuantitative[time_index].begin(); jt != betweenQuantitative[time_index].end(); jt++) {
+            size_t feature_index = jt->first;
+            RegionalFeatureLayer tmp(time_index, feature_index, "between", "quantitative" );
+            std::vector<std::vector<double> > tmp_val = betweenQuantitative[time_index][feature_index];
+            tmp.setFeatures(tmp_val);
+            feature_layers["between"]["quantitative"][time_index-1].push_back(tmp);
+        }
+    }
+    
+    // normalize all quantitative features across time slices
+//    normalizeWithinQuantitative();
+//    normalizeBetweenQuantitative();
+    
+}
+
+void RegionalFeatures::normalizeWithinQuantitative(void) {
+    
+    double m = 0.0;
+    size_t n_elem = 0;
+    
+    // collect values to compute mean
+    for (auto it = withinQuantitative.begin(); it != withinQuantitative.end(); it++) {
```

Comment:
Note that you can also write just `for (auto& x: withingQuantitative)` , and then replace `it->` with `x.`

This is called a range-for.